src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/quota/AzureQuotaPlugin.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    @VisibleForTesting
+    int getMemoryUsage(Azure azure) {
+        List<String> sizeNamesInUse = this.getVirtualMachineSizeNamesInUse(azure);
+        Map<String, VirtualMachineSize> virtualMachineSizes = this.getVirtualMachineSizes(sizeNamesInUse, azure);
+        return this.doGetMemoryUsage(sizeNamesInUse, virtualMachineSizes);
+    }
+
+    /**
+     * Return the total memory in use by all created virtual machines
+     *
+     * @param sizeNamesInUse list of virtual machine size names
+     * @param virtalMachineSizes list of virtual machine sizes in use
+     * @return total memory in use
+     */
+    @VisibleForTesting
+    int doGetMemoryUsage(List<String> sizeNamesInUse, Map<String, VirtualMachineSize> virtalMachineSizes) {
+        Integer initialValue = NO_USAGE;
+        return sizeNamesInUse.stream()
+                .map(sizeName -> virtalMachineSizes.get(sizeName))
+                .map(virtualMachineSize -> virtualMachineSize.memoryInMB())
+                .reduce(initialValue, Integer::sum);
+    }
+
+    /**
+     * Return a map of virtual machine sizes in use. The virtual machine size name is the key.
+     * @param sizeNames a list of virtual machine size names in use
+     * @param azure the azure client
+     * @return map where
+     */
+    @VisibleForTesting
+    Map<String, VirtualMachineSize> getVirtualMachineSizes(List<String> sizeNames, Azure azure) {
+        Map<String, VirtualMachineSize> sizes = new HashMap<>();
+        this.getVirtualMachineSizes(azure)
+                .stream()
+                .filter(virtualMachineSize -> sizeNames.contains(virtualMachineSize.name()))
+                .forEach(virtualMachineSize -> sizes.put(virtualMachineSize.name(), virtualMachineSize));
+        return sizes;
+    }
+
+    /**
+     * Return the names of the sizes used by all the virtual machines created
+     * @param azure the azure client
+     * @return a list of virtual machine size names
+     */
+    @VisibleForTesting
+    List<String> getVirtualMachineSizeNamesInUse(Azure azure) {
+        return this.getVirtualMachines(azure)
+                .stream()
+                .map(virtualMachine -> virtualMachine.size().toString())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Return a list of all virtual machines created
+     * @param azure the azure client
+     * @return a list of virtual machines
+     */
+    @VisibleForTesting
+    PagedList<VirtualMachine> getVirtualMachines(Azure azure) {
+        return azure.virtualMachines().list();
+    }
+
+    /**
+     * Return a list with all virtual machine sizes of the default region
+     * @param azure the azure client
+     * @return a list of virtual machine sizes
+     */
+    @VisibleForTesting
+    PagedList<VirtualMachineSize> getVirtualMachineSizes(Azure azure) {
+        return azure.virtualMachines().sizes().listByRegion(this.defaultRegionName);
```

Comment:
What happens if the "lsitByRegion" did not find the "defaultRegionName" ?
Is it throw an exception or return a null?

Reply:
It throws an exception, i will implement the code to handle it.

Reply:
Done.